### PR TITLE
fix for copy

### PR DESCRIPTION
**Pull request by omershlo**
## Description
This is a partial solution for #3845. I was able to reproduce the issue for `copy` using: 
```
import torch
import torch.nn as nn
import syft as sy

hook = sy.TorchHook(torch)
bob = sy.VirtualWorker(hook, id="bob")

model = nn.Linear(2, 1)
model_ptr = model.send(bob)
model_copy_ptr = model_ptr.copy()
```
The suggested fix solves it. 

However, I was not been able to reproduce the error for calling `clone()` on pointer.  It seems like a different bug. 
## Affected Dependencies
For this fix to work `import torch` is required in pointer_tensor.py

## How has this been tested?
There is not much testing required. I checked the above code works and nothing else in the library is broken. 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by omershlo** [comment]
@LaRiffle please check. 
The test will not work for clone as specified above - there is another bug with cloning a model pointer. I can address it as well but I think it is better to do it as part of another PR.

File: test/generic/pointers/test_model_pointer.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    pred_local = model(data)
+
+    hook = sy.TorchHook(torch)
+    bob = sy.VirtualWorker(hook, id="bob")
+    data_bob = data.send(bob)
+    model_bob = model.send(bob)
+    model_bob_copy = model_bob.copy()
+    pred_remote = model_bob_copy(data_bob)
+    pred_remote = pred_remote.get()
```
Comment (LaRiffle) [inline test/generic/pointers/test_model_pointer.py:21]:
```suggestion
    model_bob.get()
    pred_remote = model_bob_copy(data_bob)
    pred_remote = pred_remote.get()
```

**Comment by LaRiffle** [comment]
Ok let's keep clone for a separate PR :)

File: test/generic/pointers/test_model_pointer.py (hunk 1 above)
Comment (omershlo) [reply test/generic/pointers/test_model_pointer.py:21]: The suggested change introduces an error. I investigated and this is what I think is happening: 
1) There is no straightforward way to copy a model in torch. 
2) Upon trying to copy a model in soft under the hood you actually call clone. 
3) The difference is that (from the code): Clone should keep ids unchanged, contrary to copy.
4) Therefore we are actually creating two pointers to the same location. Once we "get" the object, it releases the other pointer as well.
5) Tensor on the other hand, do allow copy(). This code will work as expected:   
```
t = torch.tensor([3, 1.0, 2])
p1 = t.send(alice)
p2 = p1.copy()
p1.get()
p2.get()
```
6) In the example from 5 If we replace the `copy()` with `clone()` the same error will occur since p2 will have the same id as p1.
